org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/impl/CompilerOptions.java

```diff
@@ -1577,9 +1578,12 @@ protected void resetDefaults() {
 
 		// by default only lines and source attributes are generated.
 		this.produceDebugAttributes = ClassFileConstants.ATTR_SOURCE | ClassFileConstants.ATTR_LINES;
-		this.complianceLevel = this.originalComplianceLevel = ClassFileConstants.JDK1_4; // by default be compliant with 1.4
-		this.sourceLevel = this.originalSourceLevel = ClassFileConstants.JDK1_3; //1.3 source behavior by default
-		this.targetJDK = ClassFileConstants.JDK1_2; // default generates for JVM1.2
+
+		// by default be compliant with first supported version
+		final long firstSupportedJdkLevel = getFirstSupportedJdkLevel();
+		this.complianceLevel = this.originalComplianceLevel = firstSupportedJdkLevel;
+		this.sourceLevel = this.originalSourceLevel = firstSupportedJdkLevel;
+		this.targetJDK = firstSupportedJdkLevel;
```

Comment:
I believe `org.eclipse.jdt.internal.compiler.impl.CompilerOptions.originalComplianceLevel` and `org.eclipse.jdt.internal.compiler.impl.CompilerOptions.originalSourceLevel` are dead variables. These were introduced for mixed mode operation of 1.4 and 1.5 projects. All uses of these variables should evaluate to compile time true or false I think. If so we can address this in a follow up ticket. (See https://bugs.eclipse.org/bugs/show_bug.cgi?id=323633)

Reply:
Added tasks above for removal & cleanup of related JDT code that uses the variables.